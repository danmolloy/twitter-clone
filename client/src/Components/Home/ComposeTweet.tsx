import { UserCircleIcon, BookOpenIcon } from '@heroicons/react/outline'
import { useMutation, gql} from '@apollo/client'
import { useState } from 'react';
import { User } from '../../types';
import { FOLLOWINGPOSTS } from './Home'

export const POST_TWEET = gql`
  mutation Mutation($writePostContent: String) {
    writePost(content: $writePostContent) {
      id
    }
  }
`;

export const ComposeTweet = (props: {currentUser: User | undefined}) => {
  const [content, setContent] = useState('')

  const [postTweet, {data, loading, error}] = useMutation(POST_TWEET, {
    variables: {
      writePostContent: content,
    },
    refetchQueries: [
      FOLLOWINGPOSTS,
      "followsTweets"
    ]
  })

  const handleSubmit = (props: any) => {
    postTweet()
      .then(()=> setContent(""))
      .then(props.refreshTweets)
  }


  if (loading) return <p>'Posting..'</p>;

  if (error) return <p>'Error posting!'</p>;

  return (
    <div id="compose-tweet" className="flex flex-col border-b border-r w-full">
      <div className="flex flex-row">
        {props.currentUser 
        ? <img src={props.currentUser.profilePic} className="w-14 h-auto mt-4 ml-3 rounded-full"/>
        : <UserCircleIcon className="w-12 h-auto mt-4 ml-3"/>
        }
        <textarea 
        id="compose-tweet-content"
        className="text-lg w-full p-2 mt-2 h-auto ml-3 focus:outline-none" 
        placeholder="What's happening?"
        value={content}
        maxLength={140}
        onChange={(e) => setContent(e.target.value)}
        />
      </div>
      {content && 
      <div className="flex flex-row ml-16 border-b py-2">
        <div className="flex flex-row items-center px-2 rounded-full">
          <BookOpenIcon className="twitter-blue h-6 w-auto"/>
          <p className="twitter-blue text-sm font-bold ml-1">{140 - content.length} / 140</p>
        </div>
      </div>}
      <div className="flex flex-row ml-16 justify-end">
      <button 
        className={content.length === 0? "tweet-btn-disabled": "tweet-btn"}
        onClick={() => content.length > 0 && handleSubmit(props)}>
        Tweet
      </button>
      </div>
    </div>
  )
}