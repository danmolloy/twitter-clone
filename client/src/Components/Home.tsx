import { SparklesIcon } from '@heroicons/react/outline'
import { useEffect } from 'react'
import { ComposeTweet } from './ComposeTweet'
import { SingleTweet } from './SingleTweet'
import { gql, useQuery } from '@apollo/client'

const FOLLOWINGPOSTS =  gql`
query Query($followsTweetsHandle: String!) {
  followsTweets(handle: $followsTweetsHandle) {
    content
    id
    postDate
    author {
      name
      handle
      profilePic
    }
    likes {
      handle
    }
    retweets {
      handle
    }
  }
}
`;

export const Home = (props: any) => {
  const { loading, error, data } = useQuery(FOLLOWINGPOSTS, {variables: {followsTweetsHandle: props.data.currentUser.handle}})

  if (loading) {
    return <p>Loading</p>
  }

  if (error) {
    return <p>Error</p>
  }

  return (
    <div id="home">
      <div className="header border-b border-gray-200 h-14 flex flex-row justify-between">
        <h2 className="text-xl font-semibold p-4">Home</h2>
        <SparklesIcon className="w-10 p-2 my-2 h-auto mr-4 hover:bg-gray-200 rounded-full " />
      </div>
      <ComposeTweet user={props.data.currentUser} />
      <div className="h-auto w-full flex flex-col mt-0">
        {data.followsTweets && 
        data.followsTweets.map((tweet: {id:string, author:{}}) => {
          return <SingleTweet tweet={tweet} key={tweet.id} user={tweet.author} currentUser={props.data.currentUser}/>
        })
        }
      </div>
    </div>
  )
}