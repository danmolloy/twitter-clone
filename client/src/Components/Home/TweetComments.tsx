import { useMutation, gql } from '@apollo/client'
import { XCircleIcon } from '@heroicons/react/outline'
import fromUnixTime from 'date-fns/fromUnixTime'
import {  useState } from 'react'
import { FOLLOWINGPOSTS } from './Home'
import { GETUSER } from '../Profile/Profile'
import { Comment, Post, User } from '../../types'

export const POST_COMMENT = gql`
  mutation Mutation($postId: String!, $text: String!) {
    newComment(postId: $postId, text: $text) {
      commentId
      author {
        name
        handle
        profilePic
      }
      time
      text
    }
  }
`;

export const TweetComments = (props: {
  tweet: Post | undefined, 
  currentUser: User | undefined, 
  close: any}) => {
  const [postComment] = useMutation(POST_COMMENT)

  const [commentText, setCommentText] = useState("")

  const handleClick = async () => {
    if (commentText.length > 0) {
      await postComment({
        variables: {
          postId: props.tweet?.id,
          text: commentText
        },
        refetchQueries: window.location.pathname === "/home"  || (/id="home/g).test(document.getElementById("app-page")!.outerHTML)
        ? [FOLLOWINGPOSTS,
          "followsTweets"]
        : [GETUSER,
        "getUserProfile"]
      })
      setCommentText("")
    }
  }

  return (
    <div id="tweet-comments" className="flex flex-col top-0 sm:top-12 border bg-white fixed items-center justify-center h-auto w-auto rounded-lg shadow-md">
      <div className="flex flex-row p-4 w-full border-b ">
        <img src={props.tweet?.author.profilePic} alt={`Profile picture of ${props.tweet?.author.name}`} className="w-14 h-auto rounded-full"/>
        <div className="ml-3 flex flex-col w-full">
          <div className="flex flex-row w-full justify-between">
            <div className="flex flex-row">
              <div className="flex flex-row flex-wrap">
                <h3 className="font-bold hover:underline">{props.tweet && props.tweet.author.name}</h3>
                <h4 className="text-gray-500 ml-1">{props.tweet && props.tweet.author.handle}</h4>
              </div>
              <span className="sm:flex hidden text-gray-500 ml-1">•</span>
              <h4 className="text-gray-500 ml-1 hover:underline">{props.tweet && String(fromUnixTime(props.tweet.postDate)).slice(0, 15)}</h4>
            </div>
          </div>
          <p>{props.tweet && props.tweet.content}</p>
        </div>
        <button onClick={() => props.close()}>
          <XCircleIcon className="w-10 p-1 h-auto text-red-500 hover:bg-red-50 rounded-full -mt-8 -mr-4 bg-white"/>
        </button>
      </div>
      <div className="flex flex-row mt-4 py-2 justify-between items-center border-b">
        <img src={props.currentUser && props.currentUser.profilePic} alt="Logged in user" className="w-14 h-auto rounded-full"/>
        <input id="comment-input" placeholder="Tweet your reply" maxLength={50} className="comment-input mx-4 h-8 w-full p-1 rounded-full border" value={commentText} onChange={(e) => setCommentText(e.target.value) }/>
        <button id="post-comment-btn" className="reply-button tweet-btn h-8" onClick={() => handleClick()}>Reply</button>
      </div>
      {props.tweet && props.tweet.comments?.map((i: Comment) => {
        return <div key={i.commentId}>
          <div className="flex flex-row border-b py-4">
            <img src={i.author.profilePic} 
            alt={`Profile picture of ${i.author.name}`} 
            className="w-14 h-auto rounded-full"/>
            <div className="ml-3 flex flex-col w-full">
              <div className="flex flex-row w-full justify-between">
                <div className="flex flex-row">
                  <div className="flex flex-row">
                    <h3 className="font-bold hover:underline">{i.author.name}</h3>
                    <h4 className="text-gray-500 ml-1">{i.author.handle}</h4>
                  </div>
                  <span className="text-gray-500 ml-1">•</span>
                  <h4 className="text-gray-500 ml-1 hover:underline">{String(fromUnixTime(Number(i.time))).slice(0, 15)}</h4>
                </div>
              </div>
              <p>{i.text}</p>
            </div>
          </div>
        </div>
      })}
    </div>
  )
}