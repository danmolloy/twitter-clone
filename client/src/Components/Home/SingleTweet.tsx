import { 
  UserCircleIcon, 
  HeartIcon, 
  RefreshIcon, 
  ChatIcon 
} from '@heroicons/react/outline'
import { Link } from 'react-router-dom'
import { gql, useMutation } from '@apollo/client';
import fromUnixTime from 'date-fns/fromUnixTime'
import { useState } from 'react';
import { User, Post } from '../../types';
import { TweetComments } from './TweetComments';
import { FOLLOWINGPOSTS } from './Home';
import { GETUSER } from "../Profile/Profile"

export const LIKE_POST = gql`
  mutation Mutation($likePostPostId: String) {
    likePost(postID: $likePostPostId) {
      likes {
        handle
      }
    }
  }
`;

export const RETWEET_POST = gql`
  mutation Mutation($retweetPostPostId: String) {
    retweetPost(postID: $retweetPostPostId) {
      retweets {
        handle
      }
    }
  }
`;


export const DELETE_POST = gql`
  mutation Mutation($postId: String) {
    deletePost(postId: $postId) {
      id
    }
  }
`;

export const FOLLOWS_UNFOLLOWS = gql`
  mutation Mutation($followHandle: String) {
    followUnfollowUser(followHandle: $followHandle) {
      name
      followers {
        handle
      }
    }
  }
`;


export const SingleTweet = (props: {
  author: User | undefined, 
  currentUser: User | undefined, 
  tweet: Post |undefined, updatePage: any}) => {
  const [showMenu, setShowMenu] = useState(false)
  const [showComments, setShowComments] = useState(false)
 
  const [likePost] = useMutation(LIKE_POST, {
    variables: {
      likePostPostId: props.tweet && props.tweet.id
    }, refetchQueries: window.location.pathname === "/home" || window.location.pathname === "/"
    ? [
      FOLLOWINGPOSTS,
      "followsTweets"
    ]
    : [ 
      GETUSER,
      "getUserProfile"
    ]
  })

  const [retweetPost] = useMutation(RETWEET_POST, {
    variables: {
      retweetPostPostId: props.tweet && props.tweet.id
    }, refetchQueries: window.location.pathname === "/home" || window.location.pathname === "/"
    ? [
      FOLLOWINGPOSTS,
      "followsTweets"
    ]
    : [ 
      GETUSER,
      "getUserProfile"
    ],
    onCompleted: () => alert("View your retweets on your profile.")
  })

  const [deletePost] = useMutation(DELETE_POST, {
    variables: {
      postId: props.tweet && props.tweet.id
    },
    refetchQueries: window.location.pathname === "/home"  || window.location.pathname === "/"
    ? [
      FOLLOWINGPOSTS,
      "followsTweets"
    ]
    : [ 
      GETUSER,
      "getUserProfile"
    ]
  })

  const [followsUnfollows] = useMutation(FOLLOWS_UNFOLLOWS, {
    variables: {
      followHandle: props.author?.handle
    },
    onCompleted: () => {window.location.reload()}
  })
  
  return (
    <div id="single-tweet" className={`border-b ${!showMenu && "hover:bg-gray-50"} flex flex-col items-center`}>
      {showComments && <TweetComments currentUser={props.currentUser} tweet={props.tweet} close={() => setShowComments(false)}/>}
      <div className="flex flex-row mt-4 w-full"> 
      <Link to={props.author && props.author.handle ? `/${props.author.handle.slice(1)}` : '/'} className="user-profile-link">
        {props.author && props.author.profilePic ?
        <img src={props.author && props.author.profilePic} className="w-14 h-auto mx-3 rounded-full"/>:
        <UserCircleIcon className="w-12 h-12 ml-3"/>}
      </Link>
      <div className="ml-3 flex flex-col w-full">
        <div className="flex flex-row w-full justify-between">
          <div className="flex flex-col sm:flex-row ">
            <Link to={props.author && props.author.handle ? `/${props.author.handle.slice(1)}` : '/'} className="user-profile-link flex flex-row flex-wrap">
              <h3 className="font-bold hover:underline">{props.author && props.author.name}</h3>
              <h4 className="text-gray-500 ml-1">{props.author && props.author.handle}</h4>
            </Link>
            <span className="hidden sm:flex text-gray-500 ml-1">•</span>
            <h4 className="text-gray-500 ml-1 hover:underline">{props.tweet && String(fromUnixTime(props.tweet.postDate)).slice(0, 15)}</h4>
          </div>
          <div
          id="tweet-options"
          onBlur={() => setTimeout(() => setShowMenu(false), 100)}
          onFocus={() => setShowMenu(true)}
          >
            {showMenu && <div id="options-button" className="shadow z-10 bg-white absolute -ml-24 w-32 rounded">
            { props.author && props.currentUser && props.currentUser.handle === props.author.handle 
            ? <button 
            id="delete-button"
            className="hover:bg-gray-50 p-2" 
            onClick={async() => {
              await deletePost();
              props.updatePage()
              }}>
              Delete Tweet
              </button> 
              : props.currentUser && props.currentUser?.follows.filter((i: any) => i.handle === props.author?.handle).length > 0
              ? <button className="unfollow-post hover:bg-gray-50 p-2" onClick={() => followsUnfollows()}>Unfollow</button>
              : <button className="follow-post hover:bg-gray-50 p-2" onClick={() => followsUnfollows()}>Follow</button>}
          </div>}
          <button 
          className="post-options text-gray-500 hover:bg-blue-50 hover:text-blue-500 rounded-full p-1 mr-2">
            •••
          </button>
          </div>
        </div>
        <div>
        <p className=" mr-2">{props.tweet && props.tweet.content}</p>
        </div>
      </div>
      </div>
      <div className="flex flex-row justify-between px-12 my-2 text-gray-500 w-full">
          <button className="show-comments flex flex-row items-center hover:text-blue-500" onClick={() => setShowComments(true)}>
            <ChatIcon className=" hover:bg-blue-50 tweet-options" />
            <p>{props.tweet && props.tweet.comments ? props.tweet.comments.length : null}</p>
          </button>
          <button 
          id="retweet-button"
          className="flex flex-row items-center hover:text-green-500"
          onClick={async () =>{
            await retweetPost();
          }}>
            <RefreshIcon className="hover:bg-green-50 tweet-options" />
            <p id="retweet-count">{props.tweet && props.tweet.retweets?.length > 0 ? props.tweet.retweets.length : null}</p>
          </button>
          <button 
          id="like-button"
          className="flex flex-row  items-center hover:text-red-500"
          onClick={async () => {
          await likePost();
          }}>
            <HeartIcon className="hover:bg-red-50 tweet-options"/>
            <p id="like-count">
              {props.tweet?.likes && props.tweet.likes.length > 0 
              ? props.tweet.likes.length 
              : null}</p>
          </button>
        </div>
    </div>
  )
}