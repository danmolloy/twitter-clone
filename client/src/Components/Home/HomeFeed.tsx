import { User, Post } from "../../types"
import { SingleTweet } from './SingleTweet'

export const HomeFeed = (props: {
  currentUser: User | undefined, 
  followsTweets: Post[] | undefined, 
  updatePage: any}) => {
  return (
    <div className="home-feed h-screen w-full flex flex-col mt-0 border-r">
      {props.followsTweets && props.followsTweets.length > 0 
      ? props.followsTweets.map((tweet: Post) => {
        return <SingleTweet 
        tweet={tweet} 
        key={tweet.id} 
        author={tweet.author} 
        currentUser={props.currentUser && props.currentUser} 
        updatePage={props.updatePage}
        />
      }) 
      : <div className="mt-4 ml-4">
          <h3 className="text-xl font-bold">No tweets to show.</h3>
          <p>Find users to follow in Explore.</p> 
        </div>}
      </div>
  )
}