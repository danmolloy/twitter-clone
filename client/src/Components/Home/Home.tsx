import { ComposeTweet } from './ComposeTweet'
import { gql, useQuery } from '@apollo/client'
import { Loading } from '../App/Loading'
import { Error } from '../App/Error'
import { User } from '../../types'
import { HomeFeed } from './HomeFeed'

export const FOLLOWINGPOSTS =  gql`
  query Query {
    followsTweets {
      content
      id
      postDate
      author {
        name
        handle
        profilePic
      }
      comments {
        commentId
        text
        time
        author {
          name
          handle
          profilePic
        }
      }
      likes {
        handle
      }
      retweets {
        handle
      }
    }
  }
`;

export const Home = (props: {currentUser: User | undefined}) => {
  const { loading, error, data, refetch } = useQuery(FOLLOWINGPOSTS)


  const updatePage = () => {
    refetch()
  }

  if (loading) {
    return <Loading />
  }

  if (error) {
    console.log(error)
    return <Error />
  }

  return (
    <div id="home border-r min-h-screen border-r">
      <div id="home-header" className="home-header border-r">
        <h2 className="h2 p-4">Home</h2>
      </div>
      <ComposeTweet currentUser={props.currentUser} />
      <HomeFeed updatePage={updatePage} currentUser={props.currentUser} followsTweets={data && data.followsTweets} />
    </div>
  )
}