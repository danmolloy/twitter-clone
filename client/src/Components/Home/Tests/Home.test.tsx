import { act } from "react-dom/test-utils";
import { unmountComponentAtNode, render } from "react-dom";
import { AUTH_TOKEN } from "../../../constants";
import { MockedProvider } from "@apollo/client/testing";
import { MemoryRouter } from "react-router";
import { ComposeTweet } from "../ComposeTweet";
import pretty from "pretty";
import { HomeMock } from "./HomeMocks";
import { Home } from "../Home";
import { userMock } from '../../App/Tests/AppTestMocks'
import { HomeFeed } from "../HomeFeed";
import { SingleTweet } from "../SingleTweet";
import { fireEvent } from "@testing-library/dom";

let container: any = null

const currentUser = userMock[0].result.data

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
  global.localStorage.setItem(AUTH_TOKEN, "auth-token");
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  global.localStorage.removeItem(AUTH_TOKEN)
})

describe("ComposeTweet component", () => {
  it("renders without error", () => {
    act(() => {
      render(
      <MockedProvider>
        <MemoryRouter>
          <ComposeTweet currentUser={undefined}/>
        </MemoryRouter>
      </MockedProvider>, container)
    })
    expect(container.textContent).toMatch(/Tweet/gi)
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  })
})

describe("Home component", () => {
  it("renders without error", async () => {
    act(() => {
      render(
        <MockedProvider mocks={HomeMock} addTypename={false}>
          <MemoryRouter>
            <Home currentUser={currentUser}/>
          </MemoryRouter>
        </MockedProvider>, container
      )
    })

    await act(async() => {
      await new Promise(resolve => setTimeout(resolve, 100))
    })

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  })
})

describe("HomeFeed component", () => {
  it("renders without error", async() => {
    act(() => {
      render(
        <MockedProvider mocks={HomeMock} addTypename={false}>
          <MemoryRouter>
            <HomeFeed
            updatePage={() => jest.fn()} 
            currentUser={currentUser} 
            followsTweets={HomeMock[0].result.data.followsTweets}/>
          </MemoryRouter>
        </MockedProvider>, container
      )
    })

    await act(async() => {
      await new Promise(resolve => setTimeout(resolve, 100))
    })

    expect(pretty(container.innerHTML)).toMatchSnapshot();

  })
})

describe("SingleTweet component", () => {
  it("renders without error", async () => {
    act(() => {
      render(
        <MockedProvider mocks={HomeMock} addTypename={false}>
          <MemoryRouter>
            <SingleTweet 
            updatePage={() => jest.fn()}
            tweet={HomeMock[0].result.data.followsTweets[0]} 
            author={HomeMock[0].result.data.followsTweets[0].author}
            currentUser={currentUser}/>
          </MemoryRouter>
        </MockedProvider>, container
      )
    })

    await act(async() => {
      await new Promise(resolve => setTimeout(resolve, 100))
    })

    expect(pretty(container.innerHTML)).toMatchSnapshot();

  })

  it("likes tweet and unlikes tweet", async () => {
    act(() => {
      render(
        <MockedProvider mocks={HomeMock} addTypename={false}>
          <MemoryRouter>
            <SingleTweet 
            updatePage={() => jest.fn()}
            tweet={HomeMock[0].result.data.followsTweets[0]} 
            author={HomeMock[0].result.data.followsTweets[0].author}
            currentUser={currentUser}/>
          </MemoryRouter>
        </MockedProvider>, container
      )
    })

    await act(async() => {
      await new Promise(resolve => setTimeout(resolve, 100))
    })
    expect(container.querySelector("#like-count").textContent).toMatch(/1/)
    fireEvent.click(container.querySelector("#like-button"))
    await act(async() => {
      await new Promise(resolve => setTimeout(resolve, 100))
    })
    expect(container.querySelector("#like-count").textContent).toMatch(/2/)

  })

  it("retweets post and unretweets post", () => {})
  it("deletes post", () => {})
  it("links to user's profile", () => {})
})

describe("TweetComment component", () => {})