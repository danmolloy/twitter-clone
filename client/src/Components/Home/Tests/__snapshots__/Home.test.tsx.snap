// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposeTweet component renders without error 1`] = `
"<div id=\\"compose-tweet\\" class=\\"flex flex-col border-b border-r w-full\\">
  <div class=\\"flex flex-row\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"w-12 h-auto mt-4 ml-3\\">
      <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M5.121 17.804A13.937 13.937 0 0112 16c2.5 0 4.847.655 6.879 1.804M15 10a3 3 0 11-6 0 3 3 0 016 0zm6 2a9 9 0 11-18 0 9 9 0 0118 0z\\"></path>
    </svg><textarea id=\\"compose-tweet-content\\" class=\\"text-lg w-full p-2 mt-2 h-auto ml-3 focus:outline-none\\" placeholder=\\"What's happening?\\" maxlength=\\"140\\"></textarea></div>
  <div class=\\"flex flex-row ml-16 justify-end\\"><button class=\\"tweet-btn-disabled\\">Tweet</button></div>
</div>"
`;

exports[`Home component renders without error 1`] = `
"<div id=\\"home border-r min-h-screen border-r\\">
  <div id=\\"home-header\\" class=\\"home-header border-r\\">
    <h2 class=\\"h2 p-4\\">Home</h2>
  </div>
  <div id=\\"compose-tweet\\" class=\\"flex flex-col border-b border-r w-full\\">
    <div class=\\"flex flex-row\\"><img class=\\"w-14 h-auto mt-4 ml-3 rounded-full\\"><textarea id=\\"compose-tweet-content\\" class=\\"text-lg w-full p-2 mt-2 h-auto ml-3 focus:outline-none\\" placeholder=\\"What's happening?\\" maxlength=\\"140\\"></textarea></div>
    <div class=\\"flex flex-row ml-16 justify-end\\"><button class=\\"tweet-btn-disabled\\">Tweet</button></div>
  </div>
  <div class=\\"h-screen w-full flex flex-col mt-0 border-r\\">
    <div id=\\"single-tweet\\" class=\\"border-b hover:bg-gray-50 flex flex-col items-center\\">
      <div class=\\"flex flex-row mt-4 w-full\\"><a href=\\"/fizzlekelly\\"><img src=\\"/fiona.png\\" class=\\"w-14 h-auto mx-3 rounded-full\\"></a>
        <div class=\\"ml-3 flex flex-col w-full\\">
          <div class=\\"flex flex-row w-full justify-between\\">
            <div class=\\"flex flex-col sm:flex-row \\"><a id=\\"user-profile-link\\" class=\\"flex flex-row flex-wrap\\" href=\\"/fizzlekelly\\">
                <h3 class=\\"font-bold hover:underline\\">Fiona Kelly</h3>
                <h4 class=\\"text-gray-500 ml-1\\">@fizzlekelly</h4>
              </a><span class=\\"hidden sm:flex text-gray-500 ml-1\\">•</span>
              <h4 class=\\"text-gray-500 ml-1 hover:underline\\">Tue Sep 28 2021</h4>
            </div>
            <div id=\\"tweet-options\\"><button class=\\"text-gray-500 hover:bg-blue-50 hover:text-blue-500 rounded-full p-1 mr-2\\">•••</button></div>
          </div>
          <div>
            <p class=\\" mr-2\\">I play the flute!</p>
          </div>
        </div>
      </div>
      <div class=\\"flex flex-row justify-between px-12 my-2 text-gray-500 w-full\\"><button class=\\"flex flex-row items-center hover:text-blue-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\" hover:bg-blue-50 tweet-options\\">
            <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M8 12h.01M12 12h.01M16 12h.01M21 12c0 4.418-4.03 8-9 8a9.863 9.863 0 01-4.255-.949L3 20l1.395-3.72C3.512 15.042 3 13.574 3 12c0-4.418 4.03-8 9-8s9 3.582 9 8z\\"></path>
          </svg>
          <p>1</p>
        </button><button id=\\"retweet-button\\" class=\\"flex flex-row items-center hover:text-green-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"hover:bg-green-50 tweet-options\\">
            <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4 4v5h.582m15.356 2A8.001 8.001 0 004.582 9m0 0H9m11 11v-5h-.581m0 0a8.003 8.003 0 01-15.357-2m15.357 2H15\\"></path>
          </svg>
          <p id=\\"retweet-count\\">1</p>
        </button><button id=\\"like-button\\" class=\\"flex flex-row  items-center hover:text-red-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"hover:bg-red-50 tweet-options\\">
            <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z\\"></path>
          </svg>
          <p id=\\"like-count\\">1</p>
        </button></div>
    </div>
  </div>
</div>"
`;

exports[`HomeFeed component renders without error 1`] = `
"<div class=\\"h-screen w-full flex flex-col mt-0 border-r\\">
  <div id=\\"single-tweet\\" class=\\"border-b hover:bg-gray-50 flex flex-col items-center\\">
    <div class=\\"flex flex-row mt-4 w-full\\"><a href=\\"/fizzlekelly\\"><img src=\\"/fiona.png\\" class=\\"w-14 h-auto mx-3 rounded-full\\"></a>
      <div class=\\"ml-3 flex flex-col w-full\\">
        <div class=\\"flex flex-row w-full justify-between\\">
          <div class=\\"flex flex-col sm:flex-row \\"><a id=\\"user-profile-link\\" class=\\"flex flex-row flex-wrap\\" href=\\"/fizzlekelly\\">
              <h3 class=\\"font-bold hover:underline\\">Fiona Kelly</h3>
              <h4 class=\\"text-gray-500 ml-1\\">@fizzlekelly</h4>
            </a><span class=\\"hidden sm:flex text-gray-500 ml-1\\">•</span>
            <h4 class=\\"text-gray-500 ml-1 hover:underline\\">Tue Sep 28 2021</h4>
          </div>
          <div id=\\"tweet-options\\"><button class=\\"text-gray-500 hover:bg-blue-50 hover:text-blue-500 rounded-full p-1 mr-2\\">•••</button></div>
        </div>
        <div>
          <p class=\\" mr-2\\">I play the flute!</p>
        </div>
      </div>
    </div>
    <div class=\\"flex flex-row justify-between px-12 my-2 text-gray-500 w-full\\"><button class=\\"flex flex-row items-center hover:text-blue-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\" hover:bg-blue-50 tweet-options\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M8 12h.01M12 12h.01M16 12h.01M21 12c0 4.418-4.03 8-9 8a9.863 9.863 0 01-4.255-.949L3 20l1.395-3.72C3.512 15.042 3 13.574 3 12c0-4.418 4.03-8 9-8s9 3.582 9 8z\\"></path>
        </svg>
        <p>1</p>
      </button><button id=\\"retweet-button\\" class=\\"flex flex-row items-center hover:text-green-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"hover:bg-green-50 tweet-options\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4 4v5h.582m15.356 2A8.001 8.001 0 004.582 9m0 0H9m11 11v-5h-.581m0 0a8.003 8.003 0 01-15.357-2m15.357 2H15\\"></path>
        </svg>
        <p id=\\"retweet-count\\">1</p>
      </button><button id=\\"like-button\\" class=\\"flex flex-row  items-center hover:text-red-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"hover:bg-red-50 tweet-options\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z\\"></path>
        </svg>
        <p id=\\"like-count\\">1</p>
      </button></div>
  </div>
</div>"
`;

exports[`SingleTweet component renders without error 1`] = `
"<div id=\\"single-tweet\\" class=\\"border-b hover:bg-gray-50 flex flex-col items-center\\">
  <div class=\\"flex flex-row mt-4 w-full\\"><a href=\\"/fizzlekelly\\"><img src=\\"/fiona.png\\" class=\\"w-14 h-auto mx-3 rounded-full\\"></a>
    <div class=\\"ml-3 flex flex-col w-full\\">
      <div class=\\"flex flex-row w-full justify-between\\">
        <div class=\\"flex flex-col sm:flex-row \\"><a id=\\"user-profile-link\\" class=\\"flex flex-row flex-wrap\\" href=\\"/fizzlekelly\\">
            <h3 class=\\"font-bold hover:underline\\">Fiona Kelly</h3>
            <h4 class=\\"text-gray-500 ml-1\\">@fizzlekelly</h4>
          </a><span class=\\"hidden sm:flex text-gray-500 ml-1\\">•</span>
          <h4 class=\\"text-gray-500 ml-1 hover:underline\\">Tue Sep 28 2021</h4>
        </div>
        <div id=\\"tweet-options\\"><button class=\\"text-gray-500 hover:bg-blue-50 hover:text-blue-500 rounded-full p-1 mr-2\\">•••</button></div>
      </div>
      <div>
        <p class=\\" mr-2\\">I play the flute!</p>
      </div>
    </div>
  </div>
  <div class=\\"flex flex-row justify-between px-12 my-2 text-gray-500 w-full\\"><button class=\\"flex flex-row items-center hover:text-blue-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\" hover:bg-blue-50 tweet-options\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M8 12h.01M12 12h.01M16 12h.01M21 12c0 4.418-4.03 8-9 8a9.863 9.863 0 01-4.255-.949L3 20l1.395-3.72C3.512 15.042 3 13.574 3 12c0-4.418 4.03-8 9-8s9 3.582 9 8z\\"></path>
      </svg>
      <p>1</p>
    </button><button id=\\"retweet-button\\" class=\\"flex flex-row items-center hover:text-green-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"hover:bg-green-50 tweet-options\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4 4v5h.582m15.356 2A8.001 8.001 0 004.582 9m0 0H9m11 11v-5h-.581m0 0a8.003 8.003 0 01-15.357-2m15.357 2H15\\"></path>
      </svg>
      <p id=\\"retweet-count\\">1</p>
    </button><button id=\\"like-button\\" class=\\"flex flex-row  items-center hover:text-red-500\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"hover:bg-red-50 tweet-options\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z\\"></path>
      </svg>
      <p id=\\"like-count\\">1</p>
    </button></div>
</div>"
`;
